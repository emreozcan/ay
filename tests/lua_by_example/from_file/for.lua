for i=1, 5, 1 do
    print('first: ', i)
end

print('loop variable scope is local:', i) -- will print nil

print()

for i=1, 10, 1 do
    print('brk: ', i)
    if i == 5 then
        break
    end
end

print()

arr = {'a', 'b', 'c', 'd'}
for i,v in ipairs(arr) do
    print('arr: i, v', i, v)
end

print()

for i=1, 5 do
  if i % 2 == 0 then goto continue end
  print('odd: ', i)
  ::continue::
end