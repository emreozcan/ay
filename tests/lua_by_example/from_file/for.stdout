first: 	1
first: 	2
first: 	3
first: 	4
first: 	5
loop variable scope is local:	nil

brk: 	1
brk: 	2
brk: 	3
brk: 	4
brk: 	5

arr: i, v	1	a
arr: i, v	2	b
arr: i, v	3	c
arr: i, v	4	d

odd: 	1
odd: 	3
odd: 	5
